### PR TITLE
Some fixes when using MADGRAD (Softplus and Stable weight decay)

### PR DESCRIPTION
**Pull request by TheZothen**
Hi,

While studying the Ranger21 code with MADGRAD as the main optimizer, I asked myself the following questions :
- Is Softplus transformation implemented with MADGRAD in your framework ?
- How do you perform stable weight decay with MADGRAD ?

I think you tested softplus transformation with AdamW as main optimizer but not with MADGRAD so I fixed it (I made the assumption that beta=50 is still the optimal beta for softplus).
For Stable weight decay, I used a cube root instead of a square root in order to have the same range of value for theta(t-1) and m(t).
I also replaced `len(list(x.size()))` by `x.dim()` in the gradient centralization part (shorter and a bit faster).

I ran some tests with a CNN architecture on a split of ESC-50 dataset (spectrograms) to ensure I broke nothing :
Capture d’écran 2021-07-18 à 17 52 33
Capture d’écran 2021-07-18 à 17 52 11
The orange curve is ranger21 without modifications.
The blue one is with the sofplus modification.
The red one is with the softplus modification and the stable weight decay using cubic root.
It is not clear if it really helps but it seems that both modification improve convergence.
I kept the same configuration between runs which is the following :
```yaml
lr: 0.0007
lookahead_active: False
lookahead_mergetime: 5
lookahead_blending_alpha: 0.5
lookahead_load_at_validation: False
use_madgrad: True
use_adabelief: False
softplus: True
using_gc: True
using_normgc: True
gc_conv_only: False
normloss_active: True
normloss_factor: 1e-4
use_adaptive_gradient_clipping: True
agc_clipping_value: 1e-2
agc_eps: 1e-3
momentum_type: pnm
pnm_momentum_factor: 1.0
momentum: 0.9
eps: 1e-8
num_batches_per_epoch: 26
num_epochs: 250
use_cheb: False
use_warmup: False
num_warmup_iterations: None
warmdown_active: False
warmdown_start_pct: 0.72
warmdown_min_lr: 1e-5
weight_decay: 1e-4
decay_type: stable
warmup_type: linear
warmup_pct_default: 0.22
logging_active: True
```
I hope it will help you in your investigations.

PS : I love your work, I also tried to modify MADGRAD and thanks to you I discovered great new papers on optimizers.

### CODE REVIEW

**Comment by lessw2020** [comment]
Hi @TheZothen,
Awesome, thanks a ton for extending things on the madgrad side!  
You are correct - most of the focus has been on AdamW as the main engine, in part b/c it was doing so well as-is that the madgrad side got pushed off until after our paper.

**Comment by lessw2020** [comment]
Reviewed the changes and merged, thanks again!  I'll add you into the credits on the codebase header comments as well.
